projects/igniteui-angular/src/lib/core/styles/components/icon-button/_icon-button-theme.scss

```diff
@@ -230,35 +229,46 @@
             color: var-get($flat-theme, 'hover-foreground');
         }
 
-        &:focus-visible {
-            background: var-get($flat-theme, 'focus-background');
-            color: var-get($flat-theme, 'focus-foreground');
-            border-color: var-get($flat-theme, 'focus-border-color');
-
-            &:hover {
-                background: var-get($flat-theme, 'focus-hover-background');
-                color: var-get($flat-theme, 'focus-hover-foreground');
+        @if ($variant == 'material') {
```

Comment:
Why are you using different variables for the active state in different themes? In the Material theme it's "focus-background," while in the other themes, it's "active-background." I think It would be better to change the color of the active-background/foreground in the schema instead of using different variables.

Reply:
This is needed because only in the Material theme, the active background/foreground are not the same as the focused active background/foreground. If we want to get rid of these if statements, we have to create additional two properties for all themes and variables.